src/otx/core/data/entity/instance_segmentation.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTX instance segmentation data entities."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from otx.core.types.task import OTXTaskType
+
+from .base import OTXBatchDataEntity, OTXBatchPredEntity, OTXDataEntity, OTXPredEntity
+
+if TYPE_CHECKING:
+    from datumaro import Polygon
+    from torch import LongTensor
+    from torchvision import tv_tensors
+
+
+@dataclass
+class InstanceSegDataEntity(OTXDataEntity):
+    """Data entity for instance segmentation task.
+
+    Attributes:
+        bboxes (tv_tensors.BoundingBoxes): The bounding boxes of the instances.
+        masks (tv_tensors.Mask): The masks of the instances.
+        labels (LongTensor): The labels of the instances.
+        polygons (list[Polygon]): The polygons of the instances.
+    """
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX Task type definition."""
+        return OTXTaskType.INSTANCE_SEGMENTATION
+
+    bboxes: tv_tensors.BoundingBoxes
+    masks: tv_tensors.Mask
+    labels: LongTensor
+    polygons: list[Polygon]
```

Comment:
I can replace `list[datumaro.Polygon]` with `list[torch.FloatTensor]`, but it's important to highlight in `MMDetInstanceSegCompatibleModel._customize_inputs` function. There's a requirement to convert torch CPU tensors to GPU tensors due to the limitation in mmdet.PolygonMasks, which exclusively accepts numpy arrays (refer to the implementation [[link](https://github.com/open-mmlab/mmdetection/blob/main/mmdet/structures/mask/structures.py#L643)]). As you pointed out, this conversion can potentially impact processing speed.

If you believe that transitioning to `list[torch.FloatTensor]` is still preferable, please advise accordingly. Alternatively, if you have any alternative suggestions, I'm open to exploring them.

Reply:
Thank you for your response.  Just to confirm, are you suggesting that we proceed with handling the optimization in the current pull request, or would you prefer to leave it as future work?

Reply:
Sorry, I missed "not" here, "However, I agree that this can 'not' be handled in this PR and let it leave as the future work.". Absolutely, it should be left as future work.